### ISSUE TITLE
error with certificate = pkcs12: unknown digest algorithm: 2.16.840.1.101.3.4.2.1

### ISSUE THREAD

**Issue by MarkusatBLM** [Azure/azure-storage-azcopy#2265]
##### Note: The version is visible when running AzCopy without any argument
azcopy version 10.19.0

### Which platform are you using? (ex: Windows, Mac, Linux)
Windows Server 2019

### What command did you run?
_azcopy login --service-principal --certificate-path "C:\xyz\cert.pfx --application-id "xxxxxxxxxxxxxxxxxx"_

### What problem was encountered?
_Failed to perform login command:
pkcs12: unknown digest algorithm: 2.16.840.1.101.3.4.2.1_

### How can we reproduce the problem in the simplest way?
That's the problem I suppose. But I'll try to attach the public BASE 64 encoded CER file, maybe the information helps.

### Have you found a mitigation/solution?
Tried to use the pem format but that gives me some other error:
_Failed to perform login command:
encrypted private key block has invalid format OR your cert password may be incorrect_

I've double checked that the cert password is not the issue.

[Export FTP Cert.zip](https://github.com/Azure/azure-storage-azcopy/files/11829508/Export.FTP.Cert.zip)

**Comment by MarkusatBLM** [comment]
Just another note to this:
_digest algorithm: 2.16.840.1.101.3.4.2.1_  is the SHA-256 algorithm. Maybe an outdated library in azcopy which can't handle newer Hash algorithm?
Adding OID Reference: [Secure Hash Algorithms](https://oidref.com/2.16.840.1.101.3.4.2)

**Comment by vibhansa-msft** [comment]
Yes, this is SHA-256 which appears to be not supported by go crypto library that we have integrated. Will mark this as a feature-request, meanwhile you can try using other digest algorithms.

**Comment by MarkusatBLM** [comment]
>appears to be not supported by go crypto library that we have integrated

This cannot be only a feature request, this is basic crypto handling and a concern of security.
I've just googled "go crypto library" and got the following match:
[Go Crypto Package](https://pkg.go.dev/crypto)

This should be critical bug IMO because azcopy doesn't support strong certificate encryption as far as I can tell. Correct me if I'm wrong

**Comment by vibhansa-msft** [comment]
@adreed-msft : can you check this is supported or not from our end.

**Comment by gapra-msft** [comment]
@vibhansa-msft and @adreed-msft this might be resolved by our Track 2 migration. Checking with azidentity team if this is supported by their client certificate credential.

**Comment by MarkusatBLM** [comment]
Is there any update on this matter?

**Comment by nakulkar-msft** [comment]
Hi @MarkusatBLM can you can use this instead of login command.
https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-authorize-azure-active-directory#authorize-a-service-principal-1

**Comment by gapra-msft** [comment]
@MarkusatBLM any update on the ask above?

**Comment by MarkusatBLM** [comment]
@gapra-msft Hi!
Thank you for the reminder. 
The workaround is functional. Nevertheless, I would still opt for the approach involving a strong certificate. It gives more control to me as a sysadmin.
